test/rpc_api_mock.go

```diff
@@ -52,6 +53,27 @@ func (mock *mockService) Unpin(ctx context.Context, in api.PinSerial, out *struc
 	return nil
 }
 
+func (mock *mockService) PinPath(ctx context.Context, in api.PinPath, out *api.PinSerial) error {
+	_, err := gopath.ParsePath(in.Path)
+	if err != nil {
+		return err
+	}
+	*out = api.PinSerial{
```

Comment:
PinWithOpts()

Reply:
It will become
```go
*out = api.PinWithOpts(MustDecodeCid(TestCidResolved), in.PinOptions).ToSerial()
```
I happen to like the current version

Reply:
Right, but you understand why using a constructor is better than using a hardcoded object?

Reply:
Already pushed https://github.com/ipfs/ipfs-cluster/blob/68e3b90417ffbad89d41a70ac81d85f9037f8848/test/rpc_api_mock.go#L61